### PR TITLE
VB-1140 Visit scheduler timing out

### PR DESCRIPTION
**Pull request by AledWEvans**
## What does this pull request do?

attempts to fix timeout by reducing calls to DB with checking non-association with visits

## What is the intent behind these changes?

to fix the bug

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt [hunk 1]
```diff
@@ -44,6 +45,10 @@ class SessionService(
   private val policyNonAssociationWholeDay: Boolean,
 ) {
 
+  companion object {
+    val LOG: Logger = LoggerFactory.getLogger(this::class.java)
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt:49]: Minor observation, we could rename this val to log to keep it consistent with other instances.

Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt:49]: Not sure as it is a constant and the convention with constants is uppercase right?

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt:49]: Ok, yes as mentioned a very minor observation.

**Comment by dhirajshettymoj** [comment]
LGTM!!

**Review by dhirajshettymoj — APPROVED** [review]
LGTM!!

**Pull request merged**